Refactoring

hj0202 opened the pull request:
We refactored your project for reusability and understandability.
For more details, please check with commit.

Thanks.

Kamuno commented:
Can you let me know when you are done, so I can review the code? I see you have continues working on it after initiating the PR. 
:) Thank you.

hj0202 commented:
I'll let you know when I'm done. Thank you

hj0202 commented:
Refactored the project for reusability and understandability.
And I added a test case to find the error.

I'm done, so please review my pull request and check commits for more details.

Thanks.

Kamuno commented:
Okay. This might take a while 😅
Thank you very much for your contribution 😄
